src/sources/util/http.rs

```diff
@@ -156,27 +189,32 @@ pub trait HttpSource: Clone + Send + Sync + 'static {
             let svc = filter
                 .and(warp::path::end())
                 .and(warp::header::optional::<String>("authorization"))
+                .and(warp::header::optional::<String>("content-encoding"))
                 .and(warp::header::headers_cloned())
                 .and(warp::body::bytes())
                 .and(warp::query::<HashMap<String, String>>())
                 .and_then(
                     move |auth_header,
+                          encoding_header,
                           headers: HeaderMap,
                           body: Bytes,
                           query_parameters: HashMap<String, String>| {
+                        let _guard=span.enter();
                         info!(message = "Handling HTTP request.", headers = ?headers, rate_limit_secs = 30);
 
                         let mut out = out.clone();
 
-                        let body_size = body.len();
-                        let events = match auth.is_valid(&auth_header) {
-                            Ok(()) => self.build_event(body, headers, query_parameters),
-                            Err(err) => Err(err),
-                        };
+                        let events = auth
+                            .is_valid(&auth_header)
+                            .and_then(|()| decode(&encoding_header, body))
+                            .and_then(|body| {
+                                self.build_event(body.clone(), headers, query_parameters)
+                                    .map(|events| (events, body.len()))
```

Comment:
```suggestion
                                let body_size = body.len();
                                self.build_event(body, headers, query_parameters)
                                    .map(|events| (events, body_size))
```
`.len()`  before `build_event` will be cheaper than `.clone()`..